src/services/firebase/index.ts

```diff
@@ -0,0 +1,257 @@
+  } catch {}
+
+  const chain = chains?.find((chain) => chain.chainId === chainId)
+  const chainName = chain?.chainName ?? `chain ${chainId}`
+
+  const shortSafeAddress = shortenAddress(address)
+
+  const historyLink = getLink(AppRoutes.transactions.history, { address, chain })
+
+  const getSafeTxHashLink = (safeTxHash: string) => getLink(AppRoutes.transactions.tx, { address, chain, safeTxHash })
+
+  if (type === WebhookType.NEW_CONFIRMATION) {
+    const { owner, safeTxHash } = data
+
+    return {
+      title: `New confirmation`,
+      body: `Safe ${shortSafeAddress} on ${chainName} has a new confirmation from ${shortenAddress(
+        owner,
+      )} on transaction ${shortenAddress(safeTxHash)}.`,
+      link: getSafeTxHashLink(safeTxHash),
+    }
+  }
+
+  if (type === WebhookType.EXECUTED_MULTISIG_TRANSACTION) {
+    const { failed, txHash, safeTxHash } = data
+
+    const shortTxHash = shortenAddress(txHash)
+
+    if (failed === 'true') {
+      return {
+        title: `Transaction failed`,
+        body: `Safe ${shortSafeAddress} on ${chainName} failed to execute transaction ${shortTxHash}.`,
+        link: getSafeTxHashLink(safeTxHash),
+      }
+    } else {
+      return {
+        title: `Transaction executed`,
+        body: `Safe ${shortSafeAddress} on ${chainName} executed transaction ${shortTxHash}.`,
+        link: getSafeTxHashLink(safeTxHash),
+      }
+    }
+  }
+
+  if (type === WebhookType.PENDING_MULTISIG_TRANSACTION) {
+    const { safeTxHash } = data
+
+    return {
+      title: `New pending transaction`,
+      body: `Safe ${shortSafeAddress} on ${chainName} has a new pending transaction ${shortenAddress(safeTxHash)}.`,
+      link: getSafeTxHashLink(safeTxHash),
+    }
+  }
+
+  if (type === WebhookType.INCOMING_ETHER || type === WebhookType.OUTGOING_ETHER) {
+    const { txHash, value } = data
+
+    const currencySymbol = chain?.nativeCurrency?.symbol ?? 'ETH'
+    const currencyValue = formatUnits(value, chain?.nativeCurrency?.decimals).toString()
+    const currencyName = chain?.nativeCurrency?.name ?? 'Ether'
+
+    const shortTxHash = shortenAddress(txHash)
+
+    if (type === WebhookType.INCOMING_ETHER) {
+      return {
+        title: `${currencyName} received`,
+        body: `Safe ${shortSafeAddress} on ${chainName} received ${currencyValue} ${currencySymbol} in transaction ${shortTxHash}.`,
+        link: historyLink,
+      }
+    }
+
+    if (type === WebhookType.OUTGOING_ETHER) {
+      return {
+        title: `${currencyName} sent`,
+        body: `Safe ${shortSafeAddress} on ${chainName} sent ${currencyValue} ${currencySymbol} in transaction ${shortTxHash}.`,
+        link: historyLink,
+      }
+    }
+  }
+
+  if (type === WebhookType.INCOMING_TOKEN || type === WebhookType.OUTGOING_TOKEN) {
+    const { tokenAddress, txHash, value } = data
+
+    let balances: SafeBalanceResponse | undefined
+
+    try {
+      balances = await getBalances(chainId, address)
```

Comment:
Just a thought:
On tx service side there is an endpoint to fetch token infos:
i.e. https://safe-transaction-goerli.safe.global/api/v1/tokens/0x61fD3b6d656F39395e32f46E2050953376c3f5Ff

Should we add this to the gateway service at some point? There are multiple examples where we fetch balances only to get some token info.

Reply:
This sounds like a great idea. Let's file a feature request for it on the gateway.